gpu/gpu_info_darwin.m

```diff
@@ -0,0 +1,11 @@
+//go:build darwin
+#include "gpu_info_darwin.h"
+
+uint64_t getRecommendedMaxVRAM()
+{
+	id<MTLDevice> device = MTLCreateSystemDefaultDevice();
```

Comment:
This is a side note but is this function callable from C? if so we might be able to do this all in pure Cgo in `gpu/gpu_darwin.go`. I haven't had a chance to test that.

Reply:
Apple only provides bindings for Swift, Objective-C and C++. It looks like we need a thin layer of Objective-C.

Reply:
Great! Thanks for checking!